File: data/sample/atcoder/abc104_c/C/impl_03.c
#include <stdio.h>

#define N	10
#define INF	0x3f3f3f3f

int min(int a, int b) { return a < b ? a : b; }

int main() {
	static int pp[N], cc[N];
	int n, s_, b, i, s, p, p_;

	scanf("%d%d", &n, &s_), s_ /= 100;
	for (i = 0; i < n; i++)
		scanf("%d%d", &pp[i], &cc[i]), cc[i] /= 100;
	p_ = INF;
	for (b = 0; b < 1 << n; b++) {
		s = 0, p = 0;
		for (i = 0; i < n; i++)
			if ((b & 1 << i) != 0)
				s += pp[i] * (i + 1) + cc[i], p += pp[i];
		if (s >= s_) {
			p_ = min(p_, p);
			continue;
		}
		for (i = n - 1; i >= 0; i--)
			if ((b & 1 << i) == 0) {
				if (s + pp[i] * (i + 1) < s_)
					s += pp[i] * (i + 1), p += pp[i];
				else {
					p += (s_ - s + i) / (i + 1);
					p_ = min(p_, p);
					break;
				}
			}
	}
	printf("%d\n", p_);
	return 0;
}

File: data/sample/atcoder/abc104_c/Go/impl_03.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var (
	sc = bufio.NewScanner(os.Stdin)
	wt = bufio.NewWriter(os.Stdout)
)

const (
	inf = 1 << 60
	MOD = 998244353
)

func main() {
	/*** INIT ***/
	defer wt.Flush()
	sc.Buffer([]byte{}, inf)
	sc.Split(bufio.ScanWords)

	d, g := InputInt(), InputInt()
	p, c := make([]int, d), make([]int, d)
	for i := 0; i < d; i++ {
		p[i], c[i] = InputInt(), InputInt()
	}

	ans := inf
	for bit := 0; bit < (1 << d); bit++ {
		score, tans := 0, 0
		for i := 0; i < d; i++ {
			if 0 < (bit>>i)&1 {
				score += 100*(i+1)*p[i] + c[i]
				tans += p[i]
			}
		}
		if g <= score {
			ans = Min(ans, tans)
			continue
		}
		if bit == 24 {
		}
		for i := d - 1; 0 <= i; i-- {
			if (bit>>i)&1 == 0 {
				point := 100 * (i + 1)
				slv := Min(p[i]-1, (g-score+point-1)/point)
				score += 100 * (i + 1) * slv
				tans += slv
				if bit == 24 {
				}
			}
			if g <= score {
				break
			}
		}
		if g <= score {
			ans = Min(ans, tans)
		}
	}
	OutInt(ans)
}

func Min(a, b int) int {
	if a < b {
		return a
	} else {
		return b
	}
}

/*** I/O ***/
func InputInt() int {
	sc.Scan()
	val, _ := strconv.Atoi(sc.Text())
	return val
}

func InputString() string {
	sc.Scan()
	return sc.Text()
}

func OutInt(val int) {
	fmt.Fprintln(wt, val)
}

func OutString(s string) {
	fmt.Fprintln(wt, s)
}
